askama_derive/src/parser.rs

```diff
@@ -761,27 +761,48 @@ named_args!(block_macro<'a>(s: &'a Syntax<'a>) <Input<'a>, Node<'a>>, do_parse!(
     })
 ));
 
-fn take_raw_block<'a>(
-    i: Input<'a>,
-    s: &'a Syntax<'a>
+// Takes longest list of characters until last provided 'syntax' element is found,
```

Comment:
I'm having a really hard time trying to understand what's going on. Can you extend the comment here with an example of what it's trying to achieve and how?